### PR TITLE
chore: icon button refactor

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
- UI optimisations
- code refactor
  - drop product size model (us ui model)
  - refactor stories
- introduce small screen show/hide util (needed elsewhere)

closes [HRZ-2357](https://spryker.atlassian.net/browse/HRZ-2357)

[HRZ-2357]: https://spryker.atlassian.net/browse/HRZ-2357?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/template/themes/design-tokens/src/storefront/index.ts [hunk 1]
```diff
@@ -37,6 +38,7 @@ export const storefrontTokens: DesignToken[] = [
     },
     ...layoutMdTokens,
     ...typographyMediumTokens,
+    ...commonTokensSmall,
```
Comment (tolerants) [inline libs/template/themes/design-tokens/src/storefront/index.ts]: `layoutMdTokens`, `typographyMediumTokens`, `commonTokensSmall`... I see two kind of inconsistency here:
- mixing Medium/Small and md/sm/lg
- different order like [domain][size]Tokens and [domain]Tokens[size]

Do we have any plans to make it similar in the future?

Comment (tobi-or-not-tobi) [reply libs/template/themes/design-tokens/src/storefront/index.ts]: Yes for sure. We like to use the same model but that is currently causing a cycle dep which why we postpone this for now

**Review by AndreyYevtukhov — DISMISSED** [review]

File: libs/base/ui/action/toggle-icon/src/toggle-icon.styles.ts [hunk 2]
```diff
@@ -33,16 +33,16 @@ export const toggleIconStyles = css`
     padding: 7px;
   }
 
-  :host([has-text][size='small']) {
+  :host([has-text][size='sm']) {
```
Comment (supproduction) [inline libs/base/ui/action/toggle-icon/src/toggle-icon.styles.ts]:
```suggestion
  :host([has-text][size=${Size.Sm}]) {
```

File: libs/base/ui/form/error-message/src/stories/custom-error.stories.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ const Template: Story<unknown> = (): TemplateResult => {
           animation-timing-function: linear;
         }
       </style>
-      <oryx-icon type="error" size="large"></oryx-icon>
+      <oryx-icon type="error" size="lg"></oryx-icon>
```
Comment (supproduction) [inline libs/base/ui/form/error-message/src/stories/custom-error.stories.ts]:
```suggestion
      <oryx-icon type="error" size=Size.Lg></oryx-icon>
```

File: libs/base/ui/form/input/src/error/stories/custom-error.stories.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ const Template: Story<Props> = ({
           animation-timing-function: linear;
         }
       </style>
-      <oryx-icon slot="error" type="error" size="large"></oryx-icon>
+      <oryx-icon slot="error" type="error" size="lg"></oryx-icon>
```
Comment (supproduction) [inline libs/base/ui/form/input/src/error/stories/custom-error.stories.ts]:
```suggestion
      <oryx-icon slot="error" type="error" size=Size.Lg></oryx-icon>
```

File: libs/base/ui/graphical/icon/src/icon.styles.ts [hunk 5]
```diff
@@ -21,11 +21,11 @@ export const styles = css`
     transition: var(--oryx-transition-time);
   }
 
-  :host([size='medium']) {
-    --_margin: 3px;
+  :host([size='md']) {
```
Comment (supproduction) [inline libs/base/ui/graphical/icon/src/icon.styles.ts]:
```suggestion
  :host([size='${Size.Md}']) {
```

File: libs/base/ui/graphical/icon/src/stories/Static/icon-sizes.stories.ts [hunk 6]
```diff
@@ -20,9 +20,9 @@ const Template: Story<unknown> = (): TemplateResult => {
     </style>
     <h1>Icon size by attribute</h1>
     <div class="icon-set">
-      <oryx-icon type=${icon} size="large"></oryx-icon>
-      <oryx-icon type=${icon} size="medium"></oryx-icon>
-      <oryx-icon type=${icon} size="small"></oryx-icon>
+      <oryx-icon type=${icon} size="lg"></oryx-icon>
```
Comment (supproduction) [inline libs/base/ui/graphical/icon/src/stories/Static/icon-sizes.stories.ts]: Please, use Size enum

File: libs/base/ui/graphical/rating/src/stories/static/sizes.stories.ts [hunk 7]
```diff
@@ -7,15 +7,15 @@ export default { title: `${storybookPrefix}/Graphical/Rating/Static` } as Meta;
 const Template: Story<unknown> = (): TemplateResult => {
   return html`
     <oryx-rating readonly value="4.3">readonly, large</oryx-rating>
-    <oryx-rating readonly value="4.3" size="small">readonly, small</oryx-rating>
+    <oryx-rating readonly value="4.3" size="sm">readonly, small</oryx-rating>
```
Comment (supproduction) [inline libs/base/ui/graphical/rating/src/stories/static/sizes.stories.ts]: Please, use Size enum

File: libs/base/ui/graphical/rating/src/styles/base.styles.ts [hunk 8]
```diff
@@ -12,7 +12,7 @@ export const ratingBaseStyles = css`
     gap: 10px;
   }
 
-  :host([size='small']) {
+  :host([size='sm']) {
```
Comment (supproduction) [inline libs/base/ui/graphical/rating/src/styles/base.styles.ts]: Please, use Size enum

File: libs/base/ui/graphical/rating/src/styles/readonly-mode.styles.ts [hunk 9]
```diff
@@ -16,12 +16,12 @@ export const ratingReadonlyStyles = css`
     background-color: var(--oryx-color-canvas-500);
   }
 
-  :host([readonly]:not([size='small'])) input {
+  :host([readonly]:not([size='sm'])) input {
```
Comment (supproduction) [inline libs/base/ui/graphical/rating/src/styles/readonly-mode.styles.ts]: Please, use Size enum

File: libs/base/ui/navigation/navigation/src/navigation.component.ts [hunk 10]
```diff
@@ -22,7 +22,7 @@ export class NavigationComponent extends LitElement {
       <oryx-image resource="logo"></oryx-image>
       <slot></slot>
       <button @click=${this.toggle} aria-label=${this.toggleButtonAriaLabel}>
-        <oryx-icon type="back" size="medium"></oryx-icon>
+        <oryx-icon type="back" size="md"></oryx-icon>
```
Comment (supproduction) [inline libs/base/ui/navigation/navigation/src/navigation.component.ts]: Please, use Size enum

File: libs/base/ui/navigation/tabs/src/stories/tabs.stories.ts [hunk 11]
```diff
@@ -50,7 +50,7 @@ const Template: Story<Props> = (props: Props): TemplateResult => {
         return html`<div slot="panels" id="n${i + 1}">
           <p>Сontent for tab ${i + 1}</p>
 
-          <oryx-button size="small">
+          <oryx-button size="sm">
```
Comment (supproduction) [inline libs/base/ui/navigation/tabs/src/stories/tabs.stories.ts]: Please, use Size enum

File: libs/base/ui/overlays/modal/src/stories/static/custom-footer.stories.ts [hunk 12]
```diff
@@ -22,10 +22,10 @@ const Template: Story = (): TemplateResult => {
         Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
         tempor incididunt ut labore et dolore magna aliqua
       </p>
-      <oryx-button slot="footer" type="primary" size="small">
+      <oryx-button slot="footer" type="primary" size="sm">
```
Comment (supproduction) [inline libs/base/ui/overlays/modal/src/stories/static/custom-footer.stories.ts]: Please, use Size enum

File: libs/base/ui/overlays/notification/src/notification.component.ts [hunk 13]
```diff
@@ -32,7 +32,7 @@ export class NotificationComponent extends LitElement {
       ${when(
         this.closable,
         () => html`
-          <oryx-icon-button size="small">
+          <oryx-icon-button size="sm">
```
Comment (supproduction) [inline libs/base/ui/overlays/notification/src/notification.component.ts]: Please, use Size enum

File: libs/base/ui/overlays/notification/src/stories/Static/notification-states.stories.ts [hunk 14]
```diff
@@ -109,7 +109,7 @@ const Template: Story = ({ backgroundColor }): TemplateResult => {
                     <span class="title"> Custom title </span>
                     <div class="content">Italic content</div>
                     <div class="buttons">
-                      <oryx-button size="small">
+                      <oryx-button size="sm">
```
Comment (supproduction) [inline libs/base/ui/overlays/notification/src/stories/Static/notification-states.stories.ts]: Please, use Size enum

File: libs/base/ui/src/utilities/model/common.ts [hunk 15]
```diff
@@ -1,7 +1,9 @@
...
-  large = 'large',
-  medium = 'medium',
-  small = 'small',
+export const enum Size {
+  Xs = 'xs',
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+  Xl = 'xl',
 }
```
Comment (supproduction) [inline libs/base/ui/src/utilities/model/common.ts:7]: I would say we should add one more reusable variable. (e.g) MainSize = [Size.Lg, Size.Md, Size.Sm] and reuse it in all cases where we need `[Size.Lg, Size.Md, Size.Sm]`

File: libs/base/ui/structure/collapsible/src/collapsible.component.spec.ts [hunk 16]
```diff
@@ -38,7 +38,7 @@ describe('CollapsibleComponent', () => {
 
       it('should have a medium sized oryx-button', () => {
         const toggle = element.shadowRoot?.querySelector('oryx-button');
-        expect(toggle?.getAttribute('size')).toBe('medium');
+        expect(toggle?.getAttribute('size')).toBe('md');
```
Comment (supproduction) [inline libs/base/ui/structure/collapsible/src/collapsible.component.spec.ts]: Please, use Size enum

File: libs/base/ui/structure/collapsible/src/collapsible.component.ts [hunk 17]
```diff
@@ -67,8 +67,8 @@ export class CollapsibleComponent
   /**
    * returns the size of the control based on the appearance of the component.
    */
-  protected get controlSize(): 'small' | 'medium' | 'large' {
-    return this.isInline ? 'small' : 'medium';
+  protected get controlSize(): 'sm' | 'md' | 'lg' {
```
Comment (supproduction) [inline libs/base/ui/structure/collapsible/src/collapsible.component.ts]:
```suggestion
  protected get controlSize(): Size {
```

File: libs/base/ui/structure/text/src/text.component.ts [hunk 18]
```diff
@@ -49,7 +49,7 @@ export class TextComponent extends LitElement implements TextProperties {
       ${when(
         !this.hideToggle,
         () => html` <slot name="toggle">
-          <oryx-icon-button size="small">
+          <oryx-icon-button size="sm">
```
Comment (supproduction) [inline libs/base/ui/structure/text/src/text.component.ts]: Please, use Size enum

File: libs/domain/cart/entry/src/entry.component.ts [hunk 19]
```diff
@@ -113,7 +113,7 @@ export class CartEntryComponent extends ComponentMixin<CartEntryOptions>() {
               ></cart-entry-confirmation>
             `,
             () => html`
-              <oryx-icon-button size="small">
+              <oryx-icon-button size="sm">
```
Comment (supproduction) [inline libs/domain/cart/entry/src/entry.component.ts]: Please, use Size enum

File: libs/domain/cart/entries/src/entries.component.ts [hunk 20]
```diff
@@ -91,7 +91,7 @@ export class CartEntriesComponent extends CartComponentMixin<CartEntriesOptions>
       <section>
         <oryx-icon type="cart"></oryx-icon>
         <p>Your shopping cart is empty</p>
-        <oryx-button size="large">
+        <oryx-button size="lg">
```
Comment (supproduction) [inline libs/domain/cart/entries/src/entries.component.ts]: Please, use Size enum

File: libs/domain/cart/totals/src/totals.component.ts [hunk 21]
```diff
@@ -145,7 +145,7 @@ export class CartTotalsComponent extends CartComponentMixin<CartTotalsComponentO
           html`
             <small class="delivery-message">
               ${ifDefined(options?.deliveryMessage)}
-              <oryx-icon-button size="small">
+              <oryx-icon-button size="sm">
```
Comment (supproduction) [inline libs/domain/cart/totals/src/totals.component.ts]: Please, use Size enum

File: libs/domain/checkout/guest/src/guest.component.ts [hunk 22]
```diff
@@ -46,7 +46,7 @@ export class CheckoutGuestComponent extends CheckoutComponentMixin<CheckoutGuest
         create an account later.
       </slot>
 
-      <oryx-button slot="footer" size="small">
+      <oryx-button slot="footer" size="sm">
```
Comment (supproduction) [inline libs/domain/checkout/guest/src/guest.component.ts]: Please, use Size enum

File: libs/domain/product/average-rating/src/average-rating.component.spec.ts [hunk 23]
```diff
@@ -99,7 +99,7 @@ describe('Average Rating', () => {
   });
 
   describe('when size options is provided', () => {
-    const size = 'small';
+    const size = 'sm';
```
Comment (supproduction) [inline libs/domain/product/average-rating/src/average-rating.component.spec.ts]: Please, use Size enum

File: libs/domain/product/card/src/card.component.ts [hunk 24]
```diff
@@ -120,7 +120,7 @@ export class ProductCardComponent extends ProductMixin(
   protected renderRating(): TemplateResult | void {
     if (this.componentOptions?.enableRating) {
       return html`<oryx-product-average-rating
-        .options=${{ size: 'small', enableCount: false }}
+        .options=${{ size: 'sm', enableCount: false }}
```
Comment (supproduction) [inline libs/domain/product/card/src/card.component.ts]: Please, use Size enum

File: libs/domain/search/box/src/box.component.ts [hunk 25]
```diff
@@ -238,7 +238,7 @@ export class SearchBoxComponent
           @oryx.typeahead=${this.onTypeahead}
           .clearIconPosition=${ClearIconPosition.NONE}
         >
-          <oryx-icon slot="prefix" type="search" size="medium"></oryx-icon>
+          <oryx-icon slot="prefix" type="search" size="md"></oryx-icon>
```
Comment (supproduction) [inline libs/domain/search/box/src/box.component.ts]: Please, use Size enum

File: libs/domain/search/facet-value-navigation/src/facet-value-navigation.component.ts [hunk 26]
```diff
@@ -62,7 +62,7 @@ export class SearchFacetValueNavigationComponent
           this.enableClearAction && this.selectedLength,
           () =>
             html`
-              <oryx-button type="text" size="small">
+              <oryx-button type="text" size="sm">
```
Comment (supproduction) [inline libs/domain/search/facet-value-navigation/src/facet-value-navigation.component.ts]: Please, use Size enum

File: libs/domain/user/address-list/src/address-list.component.ts [hunk 27]
```diff
@@ -80,7 +80,7 @@ export class AddressListComponent extends ContentMixin<AddressListItemOptions>(
 
   protected renderEmpty(): TemplateResult | void {
     return html`<slot name="empty">
-      <oryx-icon type="location" size="large"></oryx-icon>
+      <oryx-icon type="location" size="lg"></oryx-icon>
```
Comment (supproduction) [inline libs/domain/user/address-list/src/address-list.component.ts]: Please, use Size enum

File: libs/platform/auth/src/components/login/login.component.ts [hunk 28]
```diff
@@ -200,7 +200,7 @@ export class AuthLoginComponent
 
         ${this.renderLoginOptions()}
 
-        <oryx-button size="small">
+        <oryx-button size="sm">
```
Comment (supproduction) [inline libs/platform/auth/src/components/login/login.component.ts]: Please, use Size enum

File: libs/platform/auth/src/components/logout/stories/logout.stories.ts [hunk 29]
```diff
@@ -30,7 +30,7 @@ const Template: Story<LogoutOptions> = (props): TemplateResult => {
   };
 
   return html`
-    <oryx-button size="small" @click=${(e: Event) => toggleLogin(e)}>
+    <oryx-button size="sm" @click=${(e: Event) => toggleLogin(e)}>
```
Comment (supproduction) [inline libs/platform/auth/src/components/logout/stories/logout.stories.ts]: Please, use Size enum

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/search/box/src/controllers/query-controls.controller.ts [hunk 30]
```diff
@@ -30,7 +30,7 @@ export class QueryControlsController {
         </button>
       </oryx-button>
 
-      <oryx-icon-button slot="suffix" size="small">
+      <oryx-icon-button slot="suffix" size="sm">
```
Comment (supproduction) [inline libs/domain/search/box/src/controllers/query-controls.controller.ts]: Please, use Size enum

File: libs/domain/search/facet-value-navigation/src/facet-value-navigation.component.ts [hunk 31]
```diff
@@ -84,7 +85,7 @@ export class SearchFacetValueNavigationComponent
       ${when(
         this.enableToggle,
         () => html`<div class="controls">
-          <oryx-button type="text" size="large">
+          <oryx-button type="text" size="lg">
```
Comment (supproduction) [inline libs/domain/search/facet-value-navigation/src/facet-value-navigation.component.ts]: Please, use Size enum

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/search/box/src/controllers/query-controls.controller.ts [hunk 32]
```diff
@@ -30,7 +31,7 @@ export class QueryControlsController {
         </button>
       </oryx-button>
 
-      <oryx-icon-button slot="suffix" size="small">
+      <oryx-icon-button slot="suffix" size=${Size.Lg}>
```
Comment (supproduction) [inline libs/domain/search/box/src/controllers/query-controls.controller.ts]: is it correct that we changes `small` into Size.Lg?

File: libs/domain/search/facet-value-navigation/src/facet-value-navigation.component.ts (hunk 31 above)
Comment (supproduction) [inline libs/domain/search/facet-value-navigation/src/facet-value-navigation.component.ts]:
```suggestion
          <oryx-button type="text" size=${Size.Lg}>
```

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/search/box/src/controllers/query-controls.controller.ts (hunk 32 above)
Comment (tobi-or-not-tobi) [reply libs/domain/search/box/src/controllers/query-controls.controller.ts]: no, mistake

**Review by supproduction — DISMISSED** [review]

File: libs/base/ui/graphical/icon/src/icon.styles.ts [hunk 33]
```diff
@@ -1,4 +1,8 @@
-import { css } from 'lit';
+import { Size } from '@spryker-oryx/ui';
+import { css, unsafeCSS } from 'lit';
+
+const smallSize = unsafeCSS(`[size='${Size.Sm}']`);
```
Comment (Sinucid) [inline libs/base/ui/graphical/icon/src/icon.styles.ts:4]: What about move this to the `@spryker-oryx/ui` and just import them?

Comment (tobi-or-not-tobi) [reply libs/base/ui/graphical/icon/src/icon.styles.ts:4]: I did not want to make it public actually

**Review by supproduction — APPROVED** [review]

**Pull request merged**